test/JDBC/input/BABEL_4302.sql

```diff
@@ -0,0 +1,2303 @@
+SELECT datepart(dy,'2016-02-03 13:30:05.523456');
+GO
+
+SET datefirst 1;
+GO
+SELECT datepart(dy,'2016-02-03 13:30:05.523456');
+GO
+
+SET datefirst 2;
+GO
+SELECT datepart(dy,'2016-02-03 13:30:05.523456');
+GO
+
+SET datefirst 3;
+GO
+SELECT datepart(dy,'2016-02-03 13:30:05.523456');
+GO
+
+SET datefirst 4;
+GO
+SELECT datepart(dy,'2016-02-03 13:30:05.523456');
+GO
+
+SET datefirst 5;
+GO
+SELECT datepart(dy,'2016-02-03 13:30:05.523456');
+GO
+
+SET datefirst 6;
+GO
+SELECT datepart(dy,'2016-02-03 13:30:05.523456');
+GO
+
+-- Queries that should throw error
+
+SELECT DATEPART(YEAR, cast('12:10:30.123' as time))
+GO
+
+SELECT DATEPART(QUARTER, cast('12:10:30.123' as time))
+GO
+
+SELECT DATEPART(MONTH, cast('12:10:30.123' as time))
+GO
+
+SELECT DATEPART(DAY, cast('12:10:30.123' as time))
+GO
+
+SELECT DATEPART(dow, cast('12:10:30.123' as time))
+GO
+
+SELECT DATEPART(doy, cast('12:10:30.123' as time))
+GO
+
+-- Max Limits checked
+
+SELECT DATEPART(day, CAST(9999999999 AS bit));
+GO
+
+SELECT DATEPART(day, CAST(214748 AS smallmoney));
+GO
+
+SELECT DATEPART(day, CAST(214748.3647 AS smallmoney));
+GO
+
+SELECT DATEPART(day, CAST(2958463 AS int));
+GO
+
+SELECT DATEPART(day, CAST(2958464 AS int));
+GO
+
+SELECT DATEPART(day, CAST(32767 AS smallint));
+GO
+
+SELECT DATEPART(day, CAST(32768 AS smallint));
+GO
+
+SELECT DATEPART(day, CAST(2958463 AS bigint));
+GO
+
+SELECT DATEPART(day, CAST(2958464 AS bigint));
+GO
+
+SELECT DATEPART(day, CAST(255 AS tinyint));
+GO
+
+SELECT DATEPART(day, CAST(256 AS tinyint));
+GO
+
+SELECT DATEPART(day, CAST(2958463 AS decimal));
+GO
+
+SELECT DATEPART(day, CAST(2958464 AS decimal));
+GO
+
+SELECT DATEPART(day, CAST(2958463 AS real));
+GO
+
+SELECT DATEPART(day, CAST(2958464 AS real));
+GO
+
+SELECT DATEPART(day, CAST(2958463 AS numeric));
+GO
+
+SELECT DATEPART(day, CAST(2958464 AS numeric));
+GO
+
+SELECT DATEPART(day, CAST(2958463.6 AS float));
+GO
+
+SELECT DATEPART(day, CAST(2958464.6 AS float));
+GO
+
+-- Min Limits checked
+
+SELECT DATEPART(day, CAST(-999999999 AS bit));
+GO
+
+SELECT DATEPART(day, CAST(-53690 AS smallmoney));
+GO
+
+SELECT DATEPART(day, CAST(-53691 AS smallmoney));
+GO
+
+SELECT DATEPART(day, CAST(-53690 AS int));
+GO
+
+SELECT DATEPART(day, CAST(-53691 AS int));
+GO
+
+SELECT DATEPART(day, CAST(-32768 AS smallint));
+GO
+
+SELECT DATEPART(day, CAST(-32769 AS smallint));
+GO
+
+SELECT DATEPART(day, CAST(-53690 AS bigint));
+GO
+
+SELECT DATEPART(day, CAST(-53691 AS bigint));
+GO
+
+SELECT DATEPART(day, CAST(0 AS tinyint));
+GO
+
+SELECT DATEPART(day, CAST(-1 AS tinyint));
+GO
+
+SELECT DATEPART(day, CAST(-53690 AS decimal));
+GO
+
+SELECT DATEPART(day, CAST(-53691 AS decimal));
+GO
+
+SELECT DATEPART(day, CAST(-53690 AS real));
+GO
+
+SELECT DATEPART(day, CAST(-53691 AS real));
+GO
+
+SELECT DATEPART(day, CAST(-53690 AS numeric));
+GO
+
+SELECT DATEPART(day, CAST(-53691 AS numeric));
+GO
+
+SELECT DATEPART(day, CAST(-53690 AS float));
+GO
+
+SELECT DATEPART(day, CAST(-53691 AS float));
```

Comment:
we still need to have test cases which covers min/max, NULL for all the data types and also with random field argument